docs/specs.md

```diff
@@ -87,181 +113,323 @@ can be considered as _a collection of text chunks_. How those chunks are
+
+The following sections are defined:
+ * `document`: contains relevant information (metadata) describing this
+   particular document; a non-exahustive list of possible fields (all of them
+   optional except `id`) is:
+    - `id`: an arbitrary string acting as an identifier for the document. If
+	  the document belongs to a dataset, it should be unique across all
+	  documents in the dataset.
+    - `type`: the model type for the document (_sequence_, _tree_, _table_)
+      The default value, when the field is not present, is _sequence_
+    - `date`: an ISO 8601 date defining the creation date of the document
+    - `main_lang`: an ISO 639-1 code describing the language the document is
+	  in.
+      For documents containing more than one language, it should express the
+      most frequent language in the document.
+    - `other_lang`: a list of ISO 639-1 codes indicating other languages
+	  present in the document
+    - `country`: a list of one or more ISO 3166-1 country codes for which the
+      document may be applicable (this might be used to further define the
+      applicability of country-dependent PII elements)
+    - `title`: a title for the document
+ * `dataset`: metadata describing the whole dataset this document belongs to
+   (so their fields are applicable to _all_ documents in the dataset).
+
+Additional sections may be defined for specific document types.
+
+
+## Iteration interfaces for source documents
+
+Once the Source Document has been specified, we can define program interfaces
+to _iterate_ over a document, producing document chunks. Two different iteration
+interfaces have been defined for this data structure:
+ * **structural iteration**: this is the native iteration for the document,
+   which produces elements revealing its intrinsic data structure (therefore the
+   objects delivered in the iteration may differ across document types,
+   _sequence_, _table_, _tree_)
+ * **full iteration** (or flat iteration): this iteration interface flattens the
+   document structure and always delivers a linear sequence of chunks,
+   regardless of the document type.
+
+An additional difference between the two iteration modes appears in the 
+content of the contextual information delivered within each chunk:
+ * In structural iteration, the context fields for a chunk will not contain any
+   information regarding the rest of the document (since the structure is
+   already given by the iteration), only semantic information inherent to the
+   chunk
+ * A chunk context in full iteration, however, may contain context fields that
+   refer to other chunks or to the full document (depending on its structure).
+
+The following sections detail these iteration interfaces for each document type.
+
+
+### Sequence documents
+
+#### Structural iteration
+
+A sequence document is divided into independent, adjacent chunks. Each chunk
+could be conceptually considered as one document paragraph (though that split is
```

Comment:
There is the concept of layout which is 2D, where the "adjacent" paragraphs are actually in 2 dimension. In actuallity, in hpertext, adjacent documents might be a graph. But this type of information is probably in a context of the chunk. There is implicity adjacency information from say the sequence of iteration itself. Or are we saying that iteration might not be guranteed to be linear in the structural interation case?

Reply:
If I understand correctly, this example will fall into one of the other document types (tree/table).

Reply:
Right, we have 3 abstractions, and some documents will not feet neatly into one of them. The _tree_ abstraction would be the closest one _as a document_. But hypertext links are defined (mostly) _across_ documents, and this type of context is something we haven't touched on yet,